javascript/ql/test/library-tests/TaintTracking/xml.js

```diff
@@ -34,4 +34,27 @@
     parser.write(source());
     parser.end();
 
-})();
\ No newline at end of file
+})();
+
+(function () {
+    var libxml = require("libxmljs");
+    var xml = source();
+    var xmlDoc = libxml.parseXmlString(xml);
+    var gchild = xmlDoc.get('//grandchild');
+    sink(gchild.text()); // NOT OK
+
+    var children = xmlDoc.root().childNodes();
+    var child = children[0];
+
+    sink(child.attr('foo').value()); // NOT OK
+
+    var child2 = xmlDoc.root().child()
+    sink(child2.attr('foo').name()); // NOT OK
```

Comment:
Hmm. This is equivalent to `sink("foo")`, which is safe.. But it is probably OK for us to get a FP in this silly case: why would you ever do `.attr(x).name()`?

A less silly variant, of this is `attr.name()` for a document that has been checked with a schema. But that is probably extremely rare in practice.

Reply:
You're right, the example is silly, nicely spotted.   
I think it's fine to keep anyway. The `.name()` method generally returns tainted values.